### PR TITLE
Added js3 week1 homework

### PR DESCRIPTION
**Pull request by sandhana14**

### CODE REVIEW

File: javascript/javascript3/week1/weather.js [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+if (!localStorage.getItem("city")) {
+    localData();
+} else {
+    let city = localStorage.getItem("city");
+    url =
+        "https://api.openweathermap.org/data/2.5/weather?q=" +
+        city +
+        "&appid=7eb2adb0ae663ad8be6e770a4b1a937e";
+    getWeatherInformation();
+}
```
Comment (ccshah298) [inline javascript/javascript3/week1/weather.js:129]: Great work Sandhana.. clean code with good variable name...👍

Comment (ccshah298) [reply javascript/javascript3/week1/weather.js:129]: I think one json file missing here for this "Create your own json file" exercise ... :)

Comment (ccshah298) [reply javascript/javascript3/week1/weather.js:129]: if possible then use const instead of let or var ..

Comment (sandhana14) [reply javascript/javascript3/week1/weather.js:129]: Thanks for your comments Charmi. I will make the required changes as per the request, but I have the create your own son file exercise in line 38 and the file name is main.json.Thank you again

File: javascript/javascript3/week1/weather.js [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+const loading = document.getElementsByClassName("loading_text");
+let url;
+
+
+
+function getWeatherInformation() {
+    for (var i = 0; i < loading.length; i += 1) {
+        loading[i].style.display = "block";
+    }
+    setTimeout(() => {
```
Comment (aleksd-dfds) [inline javascript/javascript3/week1/weather.js:24]: why are you doing setTimeout here?

File: javascript/javascript3/week1/weather.js [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+        for (var i = 0; i < loading.length; i += 1) {
+            loading[i].style.display = "none";
+        }
+        fetch(url)
+            .then((response) => response.json())
+            .then((weather_data) => {
+                cityNameToDisplay.innerHTML = weather_data.name;
```
Comment (aleksd-dfds) [inline javascript/javascript3/week1/weather.js:31]: very long sequence of statements here, beginning to be unreadable. Consider moving this part out into its own function (or maybe even several functions).

File: javascript/javascript3/week1/weather.js [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+
+function geoLocation() {
+    errorTag.innerHTML = "";
+    function success(position) {
+        const latitude = position.coords.latitude;
+        const longitude = position.coords.longitude;
+        localStorage.setItem("latitude", latitude);
+        localStorage.setItem("longitude", longitude);
+        url =
+            "https://api.openweathermap.org/data/2.5/weather?lat=" +
```
Comment (aleksd-dfds) [inline javascript/javascript3/week1/weather.js:98]: You're composing a string with this `api.openweathermap.org / appid=` in many places in the code. Consider having a helper function or a constant to make this easier.

File: javascript/javascript3/week1/weather.js [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+        getWeatherInformation();
+    }
+    navigator.geolocation.getCurrentPosition(success);
+}
+
+function localData() {
+    let latitude = localStorage.getItem("latitude");
+    let longitude = localStorage.getItem("longitude");
+    url =
+        "https://api.openweathermap.org/data/2.5/weather?lat=" +
```
Comment (aleksd-dfds) [inline javascript/javascript3/week1/weather.js:112]: same here

**Review by aleksd-dfds — APPROVED** [review]
OK but it would be a good exercise to simplify your code. There are long lists of statements and there's a lot of repetitive code. Try to make more use of extra variables and functions that contain common reusable functionality.

**Pull request merged**